### DISCUSSION TITLE
Can we use the account REST API? Why it is not documented?

### DISCUSSION THREAD

**Discussion by lecoqlibre** [keycloak/keycloak#10796] [Q&A] [answered]
Hi, I would like to know if I can use the Keycloak account REST API to update any user attributes? I'm wondering why I can't find any documentation about this? Is it safe to use? Thanks

**Comment by ssilvert** [comment, marked as answer]
Yes, feel free to use it. 

To make this easier, we will likely be building an account client similar to the one we have built out for the admin REST API.
https://github.com/keycloak/keycloak-nodejs-admin-client

Having such a client should help with documentation of the API.

**Reply by lecoqlibre** [reply]
Great! Thank you

**Reply by ahcamy2021** [reply]
I tried to experiment the account REST API to update user attribute using postman, but, all the time I get an error: "error": "HTTP 401 Unauthorized"

When I check the code I saw that when verifying the token the code assume that the request came from a front-End to get the baseUri.

My first question is for @ssilvert : Do you think it's an issue? And, I must be able to call this rest API from postman with a valid access token? If not, how can I proceed?

The second question is for @lecoqlibre :  Could you share your experience how do you integrate this API in your application? And, why do you opt to call this API instead of adding your features to the new account console client?

Thanks.

**Reply by Guesch** [reply]
Sorry for necroing this post, but as I struggled a bit I wanted to share a solution.

When you get your token, make sure to get it from a client which has the "manage account" scope.
The client "account-console" should have it by default (but you will need to enable "direct access grant")

curl --location 'http://localhost:8080/auth/realms/master/protocol/openid-connect/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Cookie: KEYCLOAK_LOCALE=en' \
--data-urlencode 'client_id=account-console' \
--data-urlencode 'grant_type=password' \
--data-urlencode 'username=XXXX' \
--data-urlencode 'password=XXXX'

Token received from this client will allow you to use the Management REST API as described in this blog: https://www.marcus-povey.co.uk/2020/10/12/using-the-keycloak-accounts-management-api/

Hope it can help some folks!

**Reply by luisplinares** [reply]
Hey,
I've been trying to hit the `/auth/realms/{realm}/account` endpoint, but looks like it's not working.
- From my frontend app I always get a CORS error, even though I added my frontend URL to the client Web Origins in keycloak.
- From postman, I'm getting the following error:
<img width="816" alt="image" src="https://github.com/user-attachments/assets/5fc64a17-bf87-46a5-a6eb-a414041c499c">

It looks to me that the `Account Managment API` was modified on latest verions of keycloak.

Because, I am actually able to hit this endpoint `/realms/{realm}/account` instead. I can Get and Update user info through that endpoint.

However, I am not able to update user password.
On the blog https://www.marcus-povey.co.uk/2020/10/12/using-the-keycloak-accounts-management-api/  they use this endpoint `/auth/realms/{realm}/account/credentials/password`, but I'm getting the same error on the screenshot. And if I try `/realms/{realm}/account/credentials/password` it doesn't work either

I'm using Keycloak 24.
Could it be that `/auth/realms/{realm}/account/credentials/password` and `/auth/realms/{realm}/account` endpoints were available on older verions of keycloak, but they aren't anymore?

**Reply by sh3rlock97** [reply]
did you maybe forget the header  {"accept": 'application/json'}?

**Reply by evanhurd** [reply]
I also get the CORS error when using the accounts API from another domain. The reqeust does work but its does not contain the proper CORS headers. To be clear, the OPTIONS call works fine and is not the issue. But the actually REST response does not have the "Access-Control-Allow-Origin" which then causes the browser to throw an error.

This is not a blocker for me as I can proxy the requests through a same origin domain. Sometimes it's nice not have to do that.

**Comment by sh3rlock97** [comment]
Updating this question. How can I find the different user-apis?

**Reply by Prajwalg19** [reply]
Keycloak follows openid-connect specs, so this should do
https://openid.net/specs/openid-connect-core-1_0.html
